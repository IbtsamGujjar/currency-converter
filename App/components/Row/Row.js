import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, TouchableOpacity, Text } from 'react-native';

import { Colors } from '../../../constants/colors';

const styles = StyleSheet.create({
  row: {
    paddingHorizontal: 20,
    paddingVertical: 16,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
    backgroundColor: Colors.white,
  },
  title: {
    color: Colors.text,
    fontSize: 16,
  },
  separator: {
    backgroundColor: Colors.border,
    height: StyleSheet.hairlineWidth,
    marginLeft: 20,
  },
});

export const Row = ({ title, icon, onPress }) => (
  <TouchableOpacity onPress={onPress} style={styles.row}>
    <Text style={styles.title}>{title}</Text>
    {icon}
  </TouchableOpacity>
);

Row.defaultProps = {
  title: '',
  icon: null,
  onPress: () => {},
};

Row.propTypes = {
  title: PropTypes.string,
  icon: PropTypes.element,
  onPress: PropTypes.func,
};
