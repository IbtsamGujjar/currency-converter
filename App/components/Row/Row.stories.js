import React from 'react';
import { Entypo } from '@expo/vector-icons';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import { Row } from './Row';
import { CenterView } from '../../../storybook/stories/decorator';

import { Colors } from '../../../constants/colors';

storiesOf('Row', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('Basic', () => (
    <Row
      title="Option"
      icon={<Entypo name="chevron-right" size={20} color={Colors.grey} />}
      onPress={action('tapped-default')}
    />
  ));
