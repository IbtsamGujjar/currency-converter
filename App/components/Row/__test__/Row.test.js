import React from 'react';
import renderer from 'react-test-renderer';
import { Row } from '../Row';

test('Row Snapshot', () => {
  const snapshot = renderer.create(<Row />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
