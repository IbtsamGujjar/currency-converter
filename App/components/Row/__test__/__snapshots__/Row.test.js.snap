// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#343434",
        "fontSize": 16,
      }
    }
  >
    
  </Text>
</View>
`;
